iceoryx_examples/ara_com/iox_ara_proxy.cpp

```diff
@@ -0,0 +1,135 @@
+
+#include <iostream>
+
+using namespace ara;
+using namespace iox::cxx;
+
+constexpr char APP_NAME[] = "iox-cpp-ara-proxy";
+
+int main()
+{
+    Runtime::GetInstance(APP_NAME);
+
+    iox::concurrent::smart_lock<optional<MinimalProxy>> maybeProxy;
+    optional<com::FindServiceHandle> maybeHandle;
+
+    // 1) Discover the available services
+    core::String searchString(TruncateToCapacity, "Example");
+    std::cout << "Searching for instances of '" << MinimalProxy::m_serviceIdentifier << "' called '"
+              << searchString.c_str() << "':" << std::endl;
+    auto handleContainer = MinimalProxy::FindService(searchString);
+
+    if (!handleContainer.empty())
+    {
+        // 2a) If available, create proxy from MinimalProxyHandle
+        for (auto& handle : handleContainer)
+        {
+            std::cout << "  Found instance of service: '" << MinimalProxy::m_serviceIdentifier << "', '"
+                      << handle.GetInstanceId().c_str() << "'" << std::endl;
+            maybeProxy->emplace(handle);
+        }
+    }
+    else
+    {
+        // 2b) If not available yet, setup asychronous search to be notified when the service becomes available
+        std::cout << "  Found no service(s), setting up asynchronous search with 'StartFindService'!" << std::endl;
+        auto callback = [&](com::ServiceHandleContainer<ara::com::ProxyHandleType> container,
+                            com::FindServiceHandle) -> void {
+            if (container.empty())
+            {
+                if (!maybeProxy->has_value())
+                {
+                    std::cout << "  No instance of service '" << MinimalProxy::m_serviceIdentifier
+                              << "' is available yet." << std::endl;
+                    return;
+                }
+                std::cout << "  Instance '" << maybeProxy->value().m_instanceIdentifier.c_str() << "' of service '"
+                          << MinimalProxy::m_serviceIdentifier << "' has disappeared." << std::endl;
+                maybeProxy->reset();
+                return;
+            }
+
+            for (auto& proxyHandle : container)
+            {
+                if (!maybeProxy->has_value())
+                {
+                    std::cout << "  Found instance of service: '" << MinimalProxy::m_serviceIdentifier << "', '"
+                              << proxyHandle.GetInstanceId().c_str() << "'" << std::endl;
+                    maybeProxy->emplace(proxyHandle);
+                }
+            }
+        };
+
+        auto handle = MinimalProxy::StartFindService(callback, searchString);
+        maybeHandle.emplace(handle);
+        std::cout << "  Waiting for instance called '" << searchString.c_str() << "' to become available.."
+                  << std::endl;
+    }
+
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+
+    while (!iox::posix::hasTerminationRequested())
+    {
+        {
+            auto proxyGuard = maybeProxy.getScopeGuard();
```

Comment:
`scopedLock` would maybe a better name to understand what is going on. Also for the method but this is not the right ... scope ... for this kind of change.

Reply:
You are free to open an issue ;)

Reply:
Not worth it for name changes only unless the name really does not work.